gr00t/utils/peft.py

```diff
@@ -0,0 +1,42 @@
+import torch
+from peft import LoraConfig, get_peft_model
+
+
+def _wrap_forward(model):
+    def _forward(inputs):
+        backbone_inputs, action_inputs = model.prepare_input(inputs)
+        backbone_outputs = model.backbone(backbone_inputs)
+        action_head_outputs = model.action_head(backbone_outputs, action_inputs)
+        model.validate_data(action_head_outputs, backbone_outputs, is_training=True)
+        return action_head_outputs
+
+    model.forward = _forward
+    return model
+
+
+def get_lora_model(model, rank=32, lora_alpha=16, lora_dropout=0.1):
+    target_modules = []
+
+    # Inspect model structure to find the correct paths
+    for name, module in model.named_modules():
+        # Look for linear layers in attention mechanisms
+        if isinstance(module, torch.nn.Linear):
+            if any(x in name for x in ["q_proj", "v_proj", "to_q", "to_v"]):
```

Comment:
could we do full kqv finetuning. adding `k_proj` and `to_k`?

Reply:
Nice catch!